cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+            or tag.get("data-type") == "author"
+        )
+    ]
+    # Flatten and dedupe list of judges
+    judges = ", ".join(
+        sorted(
+            list(set(itertools.chain.from_iterable(judge_list + author_list)))
+        )
+    )
+
+    judges = titlecase(judges)
+    all_data = {"judges": judges}
+    short_fields = ["attorneys", "disposition", "otherdate", "seealso"]
+    long_fields = [
+        "syllabus",
+        "summary",
+        "history",
+        "headnotes",
+        "correction",
+    ]
+
+    short_data = parse_extra_fields(soup, short_fields, False)
+
+    # Find any legal field
+    find_any_legal_field = soup.find(
+        lambda tag: tag.get("data-type") == "legal"
+    )
+
+    if find_any_legal_field:
+        # We have legal field, then collect all legal and attorneys fields
+        find_fields = soup.find_all(
+            lambda tag: tag.get("data-type") == "legal"
+            or (tag.name == "attorneys" and tag.get("data-type") is None)
+            or tag.get("data-type") == "attorneys"
+        )
+        if find_fields:
+            # Combine attorneys and legal data-type fields
+            arguments = " ".join(str(x) for x in find_fields)
+            all_data["arguments"] = arguments
+    else:
+        # Only save attorneys
+        find_attorneys_fields = soup.find_all(
+            lambda tag: tag.get("data-type") == "attorneys"
+            or (tag.name == "attorneys" and tag.get("data-type") is None)
+        )
+        if find_attorneys_fields:
+            attorneys = " ".join(str(x) for x in find_attorneys_fields)
+            all_data["attorneys"] = attorneys
+
+    if "otherdate" in short_data:
+        # Rename to correct field name
+        short_data["other_dates"] = short_data.pop("otherdate")
+
+    if "seealso" in short_data:
+        # Rename to correct field name
+        short_data["cross_reference"] = short_data.pop("seealso")
+
+    long_data = parse_extra_fields(soup, long_fields, True)
+    all_data.update(short_data)
+    all_data.update(long_data)
+    all_data = {k: v for k, v in all_data.items() if v}
+    return all_data
+
+
+def combine_non_overlapping_data(
+    cluster_id: int, harvard_data: dict
+) -> dict[str, Tuple]:
+    """Combine non overlapping data and return dictionary of data for merging
+
+    :param cluster_id: Cluster id to merge
+    :param harvard_data: The harvard data as json
+    :return: Optional dictionary of data to continue to merge
+    """
+    opinion_cluster = OpinionCluster.objects.get(id=cluster_id)
+    all_data = fetch_non_harvard_data(harvard_data)
+    changed_values_dictionary: dict[str, Tuple] = {}
+    for key, value in all_data.items():
+        cl_value = getattr(opinion_cluster, key)
+        if not cl_value:
+            # Value is empty for key, we can add it directly to the object
+            OpinionCluster.objects.filter(id=cluster_id).update(**{key: value})
```

Comment:
Won't this run one update per k/v pair? If so, that'd create a lot of records in our history table and it'd be pretty darned slow.

Reply:
yeah - it will - there is certainly a much more complicated but smarter way

Reply:
I made a change to update all the fields at the same time.